halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -84,7 +90,59 @@ func TestInsertEVMEvents(t *testing.T) {
 	}
 }
 
-func setupKeeper(t *testing.T, submissionDelay int64) (*Keeper, sdk.Context) {
+func TestDeliverDelegate(t *testing.T) {
+	t.Parallel()
+
+	submissionDelay := int64(3)
+	ethClientMock, err := ethclient.NewEngineMock(
+		ethclient.WithPortalRegister(netconf.SimnetNetwork()),
+		ethclient.WithMockSelfDelegation(k1.GenPrivKey().PubKey(), 1),
+	)
+	require.NoError(t, err)
+
+	msgServer := &msgServerMock{}
```

Comment:
nit: can do `msgServer := new(msgServerMock)`.

We avoid `foo{}` initialisations in general.